spec/support/synchronization_helpers.rb

```diff
@@ -53,11 +53,23 @@ def expect_in_thread(&block)
 
   # Defaults to 5 second timeout
   def try_wait_until(attempts: 50, backoff: 0.1)
-    loop do
+    # It's common for tests to want to run simple tasks in a background thread
+    # but call this method without the thread having even time to start.
+    #
+    # We add an extra attempt, interleaved by `Thread.pass`, in order to allow for
+    # those simple cases to quickly succeed without a timed `sleep` call. This will
+    # save simple test one `backoff` seconds sleep cycle.
+    #
+    # The total configured timeout is not reduced.
+    (attempts + 1).times do |i|
       result = yield(attempts)
       return result if result
 
-      sleep(backoff)
+      if i == 0
+        Thread.pass
+      else
+        sleep(backoff)
+      end
       attempts -= 1
 
       raise('Wait time exhausted!') if attempts <= 0
```

Comment:
This change is reasonable, although I'd suggest getting rid of the explicit `attempts -= 1` and `if attempts <= 0` as they're a bit of an artifact of the previous approach using an infinite loop.